mainsemi.c

```diff
@@ -0,0 +1,45 @@
+#include <stdio.h>
+#include "display.h"
+#include "mapio.h"
+#include "datastrc.h"
+#include "control.h"
+
+int main(int argc, char &argv){
+
+allocate_memory(*gmdt);
+//First allocate some memory
+
+allocate_pngns(*gmdt);
+//allocate memory for penguins
+
+read_data(argv[1], *gmdt);
+
+if(gmdt->crnt_player == 0){
+    generate_map(*gmdt);
+    crnt_player = 2;
+    write_data(*gmdt);
+    return 1;
+}
+
+if(gmdt->phase == 1){
+    place_penguin(*gmdt);
+
+    //update phase
+
+    write_data(*gmdt);
+    return 1;
+}
+
+if(gmdt->phase == 2){
+    get_movement(*gmdt)
+
+    //we should discuss how to pass movement info to function
+
+    move_player(*gmdt);
+
+    //move_player functions update current player number
+
+    write_data(*gmdt)
+    return 1;
+
+}
```

Comment:
All of the changes from the mainauto file would apply to this one, just not the part about renaming the functions by adding the `_ai` suffix to them - here it's not the AI doing the movements.

Also, you shouldn't have to call `get_movement()`, something like that should be included in the `move_player()` function, so that one can be removed as well.